hal-core/src/interrupt.rs

```diff
@@ -0,0 +1,125 @@
+/// An interrupt controller for a platform.
+pub trait Control {
+    type Error;
+    /// The type of this platform's interrupt vector.
+    ///
+    /// TODO(eliza): This is *probably* always a `u8`, do we really need to have
```

Comment:
i like having this open because while we could probably linearize like, two apics into one u8 "address space" of interrupts, it might be nicer to model it as a pair.

Reply:
I forgot about apic tbh. this should probably not be hardcoded to u8!